Modules/_datetimemodule.c

```diff
@@ -1961,9 +1961,11 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
 #endif
         ) {
             /* 0-pad year with century as necessary */
-            PyObject *item = PyTuple_GET_ITEM(timetuple, 0);
+            PyObject *item = PyTuple_GetItem(timetuple, 0);
```

Comment:
`PyTuple_GetItem()` raises `SystemError` if the first argument is not a tuple. This exception should not be leaked to the user, so you need to check for tuple before calling it.

Reply:
Good point. Thanks.